EIPS/rentcollector.md

```diff
@@ -0,0 +1,73 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Rent Collection for Blockchain Data
+    Author: Daniel A. Nagy <daniel@ethereum.org>, Zsolt Felfoldi <zsolt@ethereum.org>
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-11-01
+
+
+## Simple Summary
+Introducing a way to collect rent from accounts that is proportional 
+to the cost that they impose on the network by storing a large state
+over a long period of time.
+
+## Abstract
+A new global parameter called `rent` similar to `gasLimit` that can be 
+modified by miner vote is introduced. It is initialized at 0 and is 
+assumed to have been zero before its introduction. The miner validating 
+a particular block can increase rent in a bounded way at the cost of 
+forfeiting a proportional part of the mining reward. Similarly, if 
+`rent` is more than zero, one can decrease it in a bounded way and 
+receive the same additional mining reward that would need to be 
+forfeited to raise it back.
+
+Each contract account gets a new balance called `lease` used for paying 
+rent that is initialized at 0 and is assumed to have been zero before 
+its introduction. For externally controlled accounts, their `lease` 
+balance is the same as their normal balance. With each new block, an 
```

Comment:
Limiting this to one per block seems likely to make the effective probability of having rent levied extremely low. It also means that every new account created actually decreases the expected rent-per-account.

Reply:
As for the first objection, the probability would be proportional to the accounts' state in the total state. That is, if one contract had a state that takes up half the state space, it would be taxed on average in every other block. Externally controlled accounts would, indeed, have a rent levied on them extremely rarely, but that corresponds to those imposing an essentially negligible cost.
However, there is a better argument in favor of taxing more than one account in a block; assuming that nobody pays rent, there can only be one account removed per block, which would be too slow, given the huge number of accounts. In other words, deadbeat accounts would have a very long time before they get evicted.

Reply:
As for the second objection, I do not see why that would be a problem. Does it create any perverse incentives?